doc/dev-guide/src/release-process.md

```diff
@@ -1,66 +1,217 @@
-   for the reported version statements to ensure that
-   we definitely built something cleanly which reports as the right version
-   number when run `--version`.
-6. [r] Make a new PR to the [Rust Blog] adding a new release announcement post.
-7. [b/r] Ping someone in the release team to perform the actual release.
-   They can find instructions in `ci/sync-dist.py`.
-   > **Note:** Some manual testing occurs here, so hopefully they'll catch
-     anything egregious in which case abort the change and roll back.
-8. [b] Once the beta release has happened, post a new topic named "Seeking beta
-   testers for Rustup $VER_NUM" on the [Internals Forum].
-9. [r] Once the official release has happened, prepare and push a tag on the
-   latest `stable` commit.
-   - `git tag -as $VER_NUM -m $VER_NUM` (optionally without `-s` if not GPG
-     signing the tag)
-   - `git push origin $VER_NUM`
-
-[Rust Blog]: https://github.com/rust-lang/blog.rust-lang.org
-[Internals Forum]: https://internals.rust-lang.org
+2. Update `CHANGELOG.md` accordingly if necessary.
+
+Submit the changes in a PR and merge it.
+
+### 3. Update Checksum
+
+After merging the PR in the previous step, update the commit SHA checksum in
+`rustup-init.sh` to match the latest commit on `master`. Submit the change in a
+PR and merge it.
+
+### 4. Sync `master` to `stable`
+
+After merging the PR in the previous step, sync `master` to `stable` using
+`--ff-only`:
+
+```shell
+git fetch origin master:master
+git checkout stable && git merge --ff-only master
+git push origin HEAD:stable
+```
+
+This will kick off a new build on GitHub Actions, which will produce the release
+artifacts, upload them to `dev-static`, and make the new `beta` release
+available at `RUSTUP_UPDATE_ROOT=https://dev-static.rust-lang.org/rustup`.
+
+### 5. Test the `beta` Release
+
+While you wait for green CI on `stable`, double-check the functionality of
+`rustup-init.sh` and `rustup-init` just in case.
+
+Ensure all of CI is green on the `stable` branch. Once it is, check through a
+representative proportion of the builds looking for the reported version
+statements to ensure that we definitely built something cleanly which reports as
+the right version number when run `--version`.
+
+Once the beta release has happened, post a new topic named "Seeking beta testers
+for Rustup $VER_NUM" on the [Internals Forum].
+
+### 6. Prepare Blog Post for `stable` Release
+
+Make a new PR to the [Rust Blog] adding a new release announcement post.
+
+### 7. Request the `stable` Release
+
+Ping someone in the release team to perform the `stable` release.
+
+They will have to start a new CodeBuild job on AWS to run the `promote-release`
+tool for Rustup.
+
+### 8. Tag the `stable` Release
+
+Once the official release has happened, prepare and push a tag on the latest
+`stable` commit.
+
+- `git tag -as $VER_NUM -m $VER_NUM` (optionally without `-s` if not GPG
```

Comment:
Should we do this in promote release? We do for rust and cargo (also always signing the commits with the official key).